src/test/java/plugins/DisplayUrlApiTest.java

```diff
@@ -0,0 +1,160 @@
+import org.jenkinsci.test.acceptance.junit.DockerTest;
+import org.jenkinsci.test.acceptance.junit.WithDocker;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.PluginManager;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+import org.jenkinsci.test.acceptance.update_center.PluginSpec;
+import org.jenkinsci.test.acceptance.utils.mail.MailhogProvider;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.AssumptionViolatedException;
+
+import jakarta.mail.MessagingException;
+import java.io.IOException;
+import java.util.regex.Pattern;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests the display-url-api plugin.
+ */
+@WithDocker
+@Category(DockerTest.class)
+@WithPlugins({"mailer@1.18", "workflow-job", "workflow-cps", "workflow-basic-steps", "workflow-durable-task-step"})
+public class DisplayUrlApiTest extends AbstractPipelineTest {
+
+    @Inject
+    private MailhogProvider mailhogProvider;
+
+    private HttpClient client = HttpClientBuilder.create().setRedirectStrategy(new LaxRedirectStrategy()).build();
+
+    @Test
+    public void testUrlInEmail() throws IOException, MessagingException {
+        final MailhogContainer mailhog = mailhogProvider.get();
+        final WorkflowJob job = createPipelineJobWithScript(scriptForMailerPipeline(true));
+        job.startBuild().shouldFail();
+
+        String expectedUrlGeneratedByDisplayUrlApiPlugin = getURLS(job.name)[0];
+
+        mailhog.assertMail(
+                Pattern.compile("Build failed in Jenkins: .* #1"),
+                "dev@example.com",
+                Pattern.compile(expectedUrlGeneratedByDisplayUrlApiPlugin));
+
+        makeJobSucceed(job);
+        job.startBuild().shouldSucceed();
+
+        mailhog.assertMail(
+                Pattern.compile("Jenkins build is back to normal .* #2"),
+                "dev@example.com",
+                Pattern.compile(expectedUrlGeneratedByDisplayUrlApiPlugin.replace("/1/", "/2/")));
+    }
+
+    @Test
+    @WithPlugins({"blueocean"})
+    public void testRedirectsToBlueOcean() throws IOException {
+        doTestRedirects();
+    }
+
+    @Test
+    public void testRedirects() throws IOException {
+        PluginManager pm = jenkins.getPluginManager();
+        if (pm.installationStatus(new PluginSpec("blueocean")) == PluginManager.InstallationStatus.NOT_INSTALLED) {
+            doTestRedirects();
+        } else {
+            throw new AssumptionViolatedException("Test redirect with Blue Ocean has already be tested by other test on this suite");
+        }
+    }
+
+
+    private String scriptForMailerPipeline(boolean makeFail) {
+        String status = makeFail ? "FAILURE" : "SUCCESS";
+        return String.format("node {\n" +
+                "    try {\n" +
+                "        sh 'exit 1'\n" +
```

Comment:
this assumes you will always be on a `linux node` which is an incorrect assumption to make.

`sh' exit 1'`  is just to fail the build - but I think you should be able to do the exact same withouth any `node` by just calling [`error`](https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#error-error-signal)